src/Codeception/Module/Symfony2.php

```diff
@@ -143,41 +180,66 @@ public function _initialize()
+        $this->kernel->boot();
+
+        if ($this->config['cache_router'] === true) {
+            $this->persistService('router', true);
+        }
     }
 
+    /**
+     *  Initialize new client instance before each test
+     */
     public function _before(\Codeception\TestCase $test)
     {
-        $this->client = new Symfony2Connector($this->kernel);
-        $this->client->followRedirects(true);
+        $this->persistentServices = array_merge($this->persistentServices, $this->permanentServices);
+        $this->client = new Symfony2Connector($this->kernel, $this->persistentServices, $this->config['rebootable_client']);
     }
 
-    public function _getEntityManager()
+    /**
+     *  Update permanent services after each test
+     */
+    public function _after(\Codeception\TestCase $test)
     {
-        $this->bootKernel();
-        if (!$this->kernel->getContainer()->has($this->config['em_service'])) {
-            return null;
+        foreach ($this->permanentServices as $serviceName => $service) {
+            $this->permanentServices[$serviceName] = $this->grabService($serviceName);
         }
-        $this->client->persistentServices[] = $this->config['em_service'];
-        $this->client->persistentServices[] = 'doctrine.orm.default_entity_manager';
-        return $this->grabService($this->config['em_service']);
+        parent::_after($test);
     }
 
-    protected function bootKernel()
+    /**
+     * Retrieve Entity Manager.
+     *
+     * EM service is retrieved once and then that instance returned on each call
+     */
+    public function _getEntityManager()
     {
-        if ($this->kernel) {
-            return;
+        if ($this->kernel === null) {
+            $this->fail('Symfony2 platform module is not loaded');
         }
-        $this->kernel = new $this->kernelClass($this->config['environment'], $this->config['debug']);
-        $this->kernel->boot();
-        if ($this->config['cache_router'] === true) {
-            if (isset($this->permanentServices['router'])) {
-                $this->kernel->getContainer()->set('router', $this->permanentServices['router']);
-            } else {
-                $this->permanentServices['router'] = $this->grabService('router');
+        if (!isset($this->permanentServices[$this->config['em_service']])) {
+            // try to persist configured EM
+            $this->persistService($this->config['em_service'], true);
+
+            if ($this->getContainer()->has('doctrine')) {
+                $this->persistService('doctrine', true);
+            }
+            if ($this->getContainer()->has('doctrine.orm.default_entity_manager')) {
+                $this->persistService('doctrine.orm.default_entity_manager', true);
             }
         }
+        return $this->permanentServices[$this->config['em_service']];
+    }
+
+    /**
+     * Return active container.
+     *
+     * @return ContainerInterface
+     */
+    protected function getContainer()
```

Comment:
You can change visibility to `public` so users could still access container if they need.